compiler/rustc_hir_typeck/src/generator_interior/mod.rs

```diff
@@ -183,6 +192,14 @@ pub fn resolve_interior<'a, 'tcx>(
     kind: hir::GeneratorKind,
 ) {
     let body = fcx.tcx.hir().body(body_id);
+
+    // In case we are in an async block, this is the Param/Pat HirId of the
+    // `&mut Context<'_>` resume type. We can use this to explicitly prevent it
+    // from being considered as `live_across_yield`, which it is not, but the
+    // simple scope-based analysis can't tell.
+    let task_context_hid =
+        if kind == hir::GeneratorKind::Gen { None } else { Some(body.params[0].pat.hir_id) };
```

Comment:
I'd prefer to reverse the check here and see if `kind` matches any of the async generator kinds instead. Makes it less of a risk if we were to add a new generator kind or refactor something later.